### PR TITLE
feat: add asset-conversion endpoints

### PR DESCRIPTION
**Pull request by bee344**
Closes issue #1308 
## Description
Adds the endpoints `/pallets/asset-conversion/liquidity-pools` which returns an array of the existing liquidity pool tokens, and  `/pallets/asset-conversion/next-available-id` which returns a string of the next available liquidity pool's id.

## Query Params
`at`: Which block to query, defaults to latest finalized block if empty. Accepts a hash or a block number.

## Sample Response
While connected to `Westend Assst Hub` and requesting `/pallets/liquidity-pools`, the return result is/should be simmilar to the following :
```
{
   "at":{
      "hash":"0xd712c01e62c4ec702c4b56e01cbbfd1b20e9008b18f4ebee6b1d854e26a8fd82",
      "height":"91"
   },
   "pools":[
      {
         "reserves":[
            {
               "parents":"1",
               "interior":{
                  "here":null
               }
            },
            {
               "parents":"0",
               "interior":{
                  "x2":[
                     {
                        "palletInstance":"50"
                     },
                     {
                        "generalIndex":"2"
                     }
                  ]
               }
            }
         ],
         "lpToken":{
            "lpToken":"1"
         }
      },
      {
         "reserves":[
            {
               "parents":"1",
               "interior":{
                  "here":null
               }
            },
            {
               "parents":"0",
               "interior":{
                  "x2":[
                     {
                        "palletInstance":"50"
                     },
                     {
                        "generalIndex":"1"
                     }
                  ]
               }
            }
         ],
         "lpToken":{
            "lpToken":"0"
         }
      }
   ]
}
```

While connected to `Westend Asset Hub` and requesting `/pallets/next-available-id`, the return result is/should be the following :
```
{
   "at":{
      "hash":"0xedb8792b4cd1ed8a8d5638e2fa5c7739cb4f4d93a68b20d5facc4888d2753131",
      "height":"736"
   },
   "id":"2"
}
```
## TODO

- [x] Add tests

### CODE REVIEW

File: e2e-tests/historical/endpoints/asset-hub-westend/accounts/asset-balances/index.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// import accountAssetBalance450000 from './450000.json';
+// import accountAssetBalance650000 from './650000.json';
+// import accountAssetBalance960000 from './960000.json';
+// import accountAssetBalance986791 from './986791.json';
+// import accountAssetBalance1100000 from './1100000.json';
+// import accountAssetBalance1300000 from './1300000.json';
+import accountAssetBalance1800000 from './1800000.json';
+
+export const assetHubWestendAccountAssetBalanceEndpoints = [
+	// [
```
Comment (TarikGul) [inline e2e-tests/historical/endpoints/asset-hub-westend/accounts/asset-balances/index.ts:26]: There is a bunch of commented out e2e-tests here. Up to you if you think they hold merit, but just having v700 is more than enough for me atleast.

Comment (bee344) [reply e2e-tests/historical/endpoints/asset-hub-westend/accounts/asset-balances/index.ts]: Sorry, I thought I got all of them, it's done now.

File: src/services/pallets/PalletsAssetConversionService.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	async fetchNextAvailableId(hash: BlockHash): Promise<ILiquidityId> {
+		const { api } = this;
+
+		const [{ number }, id] = await Promise.all([
+			api.rpc.chain.getHeader(hash),
+			api.query.assetConversion.nextPoolAssetId(),
+		]);
+		let poolId;
```
Comment (TarikGul) [inline src/services/pallets/PalletsAssetConversionService.ts:40]: This can be a ternary: `const poolId = id.isSome ? id : 0;`

**Review by TarikGul — APPROVED** [review]
LGTM, really great job, just a few nits to resolve!

**Review by marshacb — APPROVED** [review]
Awesome PR! A very small nit/question. Once any lingering merge conflicts are handled looks good to go :)

File: src/services/test-helpers/mock/data/assetConversionEntries.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+for (let i = 0; i < assets.length; i++) {
+	const reserve = assetHubWestendRegistryV9435.createType(
+		'XcmV3MultiLocation',
+		assets[i]
+	);
+	reserves.push([native, reserve] as unknown as ITuple<
+		[XcmV3MultiLocation, XcmV3MultiLocation]
+	>);
+	console.log(reserves);
```
[the remark this answers is not in the document]
Comment (bee344) [reply src/services/test-helpers/mock/data/assetConversionEntries.ts]: Deleted it

**Comment by TarikGul** [comment]
In terms of the failing docker check in gitlab, I think its not relevant to your PR but a version of `buildah`.

**Pull request merged**